packages/teko/src/Teko_Utilities.cpp

```diff
@@ -2037,14 +2056,26 @@ const ModifiableLinearOp explicitAdd(const LinearOp & opl,const LinearOp & opr,
 
       // Build output operator
       RCP<Thyra::LinearOpBase<ST> > explicitOp;
-      if(destOp!=Teuchos::null)
+      RCP<Tpetra::CrsMatrix<ST,LO,GO,NT> > explicitCrsOp;
+      if(destOp!=Teuchos::null){
          explicitOp = destOp;
-      else
+         try {// try to reuse matrix sparsity with Add. If it fails, build new operator with add
+           RCP<Thyra::TpetraLinearOp<ST,LO,GO,NT> > tOp = rcp_dynamic_cast<Thyra::TpetraLinearOp<ST,LO,GO,NT> >(destOp);
+           explicitCrsOp = rcp_dynamic_cast<Tpetra::CrsMatrix<ST,LO,GO,NT> >(tOp->getTpetraOperator(),true);
+           Tpetra::MatrixMatrix::Add<ST,LO,GO,NT>(*tCrsOpl,transpl,scalarl,*tCrsOpr,transpr,scalarr,explicitCrsOp);
+         }
+         catch (std::exception & e) {
```

Comment:
Is it possible to know the type of the exception more precisely?

Reply:
I added a print of the exception. Is there something more useful that could be checked or displayed? Not sure we want it to print every time it fails, since it's not wrong when it fails, it just gives a new matrix with a different pointer.